hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioEventloop.java

```diff
@@ -34,49 +31,21 @@
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
 
 /**
- * Nio implementation of the {@link Eventloop}.
+ * Nio specific Eventloop implementation.
  */
-public final class NioEventloop extends Eventloop {
+class NioEventloop extends Eventloop {
 
     final Selector selector = SelectorOptimizer.newSelector();
+    private final NioReactor reactor;
```

Comment:
Are `Reactor` in parent class and `NioReactor` suppose to co-exist?

Reply:
'yes' because you need to be able to do a fast lookup from the outside and I don't want to cast if that isn't needed.